### PR TITLE
chore: use generic LRU cache

### PR DESCRIPTION
**Pull request by muXxer**
This PR upgrades golang-lru to the generics branch v2.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -51,11 +51,11 @@ require (
 	github.com/raulk/go-watchdog v1.3.0
 	github.com/stretchr/testify v1.8.1
 	go.opencensus.io v0.24.0
-	go.uber.org/fx v1.18.2
-	go.uber.org/goleak v1.1.12
-	golang.org/x/crypto v0.3.0
+	go.uber.org/fx v1.19.0
```
Comment (marten-seemann) [inline go.mod:54]: There’s something wrong with the dependencies here. There’s no reason to update fx and qtls in this PR.

**Review by marten-seemann — CHANGES_REQUESTED** [review]
This looks like a reasonable improvement, but there something wrong with the dependencies.

File: go.mod (hunk 1 above)
Comment (muXxer) [reply go.mod:54]: Should I revert the second commit, which was basically a `go get -u`?
Maybe you can update the other dependencies some other day :+1:

Comment (marten-seemann) [reply go.mod:54]: Yes, please do. It’s good to keep PRs limited to one change at a time.

Comment (muXxer) [reply go.mod]: I knew it wasn't clean when I did it, but I like to live dangerously :trollface: 
Should be fine now, I reverted the second commit :+1:

**Review by marten-seemann — APPROVED** [review]
Nice! Always happy to see generic code (being used)!

**Pull request merged**